lab_3_generate_by_ngrams/main.py

```diff
@@ -93,6 +125,29 @@ def encode(self, text: str) -> Optional[tuple[int, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or len(text) == 0:
+            return None
+
+        encoded_text = {}
+        encoded_corpus = []
+        tokenized_text = self._tokenize(text)
+
+        if not tokenized_text:
+            return None
+
+        for i in range(len(tokenized_text)):
```

Comment:
Why iterate over indices if we only use those indices to find an element in the sequence?